Lib/unittest/mock.py

```diff
@@ -68,7 +69,9 @@ def _get_signature_object(func, as_instance, eat_self):
     signature object.
     Return a (reduced func, signature) tuple, or None.
     """
-    if isinstance(func, type) and not as_instance:
+    if isinstance(func, functools.partial):
+        pass
```

Comment:
What's special about `functools.partial`? It's a bit scary to special case one particular type, when there may be other that behave like it. (also curious how this relates to my original bug report?)